### DISCUSSION TITLE
Styled component's classname did not match the server issue

### DISCUSSION THREAD

**Discussion by Pixelatex** [vikejs/vike#1619] [Discussion]
### Description

Hi,

We are trying to get an app up and running with the combination of VIKE + TS + styled + apollo.

So far I've combed through a dozen inconsistent examples and got something working: 
https://github.com/Pixelatex/vike-styled-ts

However... styled components still gives the classname doesn't match server classname error... ~but seemingly only in HMR as a prod build doesn't show this. (or warnings or suppressed there, not sure)~

Normally this is solved on other solutions such as next by passing along the ssr option to the babel compiler but none of the documentation/examples I've found so far give any indication that is required? 

### Update

I was wrong, I ran the dev build and as I feared, the warning was simply suppressed. So this issue is not actually related to HMR specifically but to vike/vite or my setup of it.

**Comment by brillout** [comment]
Is there something on Vike’s side that is blocking?

**Reply by Pixelatex** [reply]
I don't know why HMR breaks it, I was hoping someone here could help me figure that bit out

**Reply by brillout** [reply]
I’d assume this isn’t related to Vike, since HMR isn’t implemented by Vike but by Vite.

**Reply by brillout** [reply]
One thing that may help is to research whether there is a solution for other Vite-based frameworks; the solution will likely also apply to Vike (but let me know if it doesn’t as it would be Vike issue then).

**Reply by brillout** [reply]
> Fand — [Today at 9:03 AM](https://discord.com/channels/1105779924658303058/1125707822705938454/1232950161798463498)
@brillout I managed to fix my issue with styled components. I noticed the existing guide is missing a bit. Can I make a PR somewhere to update the guide?
> Magne — Today at 11:26 AM
Are you talking about this guide? https://www.alexrabin.com/blog/vite-ssr-styled-components
If so, I don't know if this person hangs out in here, so perhaps contact them directly.
Otherwise, it could be a good idea to write a proper guide directly in the doc itself https://github.com/vikejs/vike/blob/main/docs/pages/styled-components/%2BPage.mdx

@Pixelatex (i.e. Fand I guess) What was the solution? Yes that'd be great, would a PR for https://vike.dev/styled-components work? Feel free to just make a rough draft PR and we/I can polish it.

**Reply by brillout** [reply]
I guess the solution was [this commit](https://github.com/Pixelatex/vike-styled-ts/commit/7a5d73c0038558c2a91fc5a7117982932630fcac)? I went ahead and added your example to https://vike.dev/styled-components, but let me know if you object.
